src/NuGet.Core/NuGet.CommandLine.XPlat/Commands/Package/Update/PackageUpdateCommandRunner.cs

```diff
@@ -133,27 +163,96 @@ internal static async Task<int> Run(PackageUpdateArgs args, ILoggerWithColor log
     {
         if (packages is null || packages.Count == 0)
         {
-            logger.LogMinimal(Strings.Unsupported_UpgradeAllPackages, ConsoleColor.Red);
-            return (null, null);
+            return await GetAllPackagesWithUpdatesAsync(project, versionChooser, settings, logger, cancellationToken);
+        }
+
+        var sourceMapping = PackageSourceMapping.GetPackageSourceMapping(settings);
+        if (sourceMapping.IsEnabled)
+        {
+            logger.LogMinimal(Strings.Unsupported_PackageSourceMapping, ConsoleColor.Red);
+            return new List<PackageUpdateResult>();
         }
 
-        if (packages.Count > 1)
+        var packagesToUpdate = new List<PackageUpdateResult>();
+        bool hasErrors = false;
+
+        foreach (var package in packages)
         {
-            logger.LogMinimal(Strings.Unsupported_MoreThanOnePackage, ConsoleColor.Red);
-            return (null, null);
+            var result = await GetSinglePackageToUpdateAsync(package, project, versionChooser, settings, logger, suppressWarnings: false, cancellationToken);
+            if (result != null)
+            {
+                packagesToUpdate.Add(result);
+            }
+            else
+            {
+                hasErrors = true;
```

Comment:
Should we exit the for loop here because we return an empty list when `hasErrors` = true.

Reply:
Ideally, we should report all the errors so that customers can take an action only once before retrying the command. In this case, `hasErrors` is a boolean we are currently returning an empty list when this `hasErrors` is true. Hence, my suggestion to exit the loop when `hasErrors` == true. Please let me know if I am missing any functional requirement.

Reply:
`GetSinglePackageToUpdateAsync` reports specific errors for whatever caused it to return null in the first place. So, the current behaviour is what you described as ideal, and exiting early would prevent that. The call the to the method passes parameter `suppressWarnings: false`, which I think indicates this.

I can add a comment if you find it unintuitive, or if anyone else has the same question. But at this time I'd prefer not to push another commit with only this change because this PR not being merged is blocking me from my next dotnet package update task (same code needs to be modified), so I'd prefer to leave minor nitpicks to a follow up PR.